src/coreclr/nativeaot/Runtime/unix/UnixNativeCodeManager.cpp

```diff
@@ -1230,43 +1230,23 @@ bool UnixNativeCodeManager::GetReturnAddressHijackInfo(MethodInfo *    pMethodIn
         return false;
     }
 
-#ifndef TARGET_LOONGARCH64
-    PTR_uintptr_t pLR = pRegisterSet->pLR;
+    PTR_uintptr_t oldLocation = pRegisterSet->GetReturnAddressRegisterLocation();
     if (!VirtualUnwind(pMethodInfo, pRegisterSet))
     {
         return false;
     }
 
-    if (pRegisterSet->pLR == pLR)
+    if (pRegisterSet->GetReturnAddressRegisterLocation() == oldLocation)
     {
         // This is the case when we are either:
         //
-        // 1) In a leaf method that does not push LR on stack, OR
-        // 2) In the prolog/epilog of a non-leaf method that has not yet pushed LR on stack
-        //    or has LR already popped off.
+        // 1) In a leaf method that does not push ReturnAddressRegisterLocation on stack, OR
+        // 2) In the prolog/epilog of a non-leaf method that has not yet pushed ReturnAddressRegisterLocation on stack
+        //    or has ReturnAddressRegisterLocation already popped off.
```

Comment:
```suggestion
        // 1) In a leaf method that does not push return address register on stack, OR
        // 2) In the prolog/epilog of a non-leaf method that has not yet pushed return address register on stack
        //    or has return address register already popped off.
```
ReturnAddressLocation is where the return address register is pushed on the stack